### PR TITLE
Simplify AlphanumericComparer, reduce code duplication

### PR DESCRIPTION
**Pull request by JustAMan**
**Changes**

Reduce the amount of copy-paste, simplify code a bit

**Issues**

N/A

### CODE REVIEW

File: MediaBrowser.Controller/Entities/BaseItem.cs [hunk 1]
```diff
@@ -387,15 +387,12 @@ public virtual string FileNameWithoutExtension
...
-                {
-                    break;
-                }
                 char thisCh = s1[thisMarker];
 
                 var thisChunk = new StringBuilder();
+                bool isNumeric = char.IsDigit(thisCh);
 
-                while ((thisMarker < s1.Length) && (thisChunk.Length == 0 || SortHelper.InChunk(thisCh, thisChunk[0])))
+                while ((thisMarker < s1.Length) && (char.IsDigit(thisCh) == isNumeric))
```
Comment (cvium) [inline MediaBrowser.Controller/Entities/BaseItem.cs]:
```suggestion
                while (thisMarker < s1.Length && char.IsDigit(thisCh) == isNumeric)
```

File: MediaBrowser.Controller/Sorting/AlphanumComparator.cs [hunk 2]
```diff
@@ -31,8 +31,9 @@ public static int CompareValues(string s1, string s2)
 
                 var thisChunk = new StringBuilder();
                 var thatChunk = new StringBuilder();
+                bool thisNumeric = char.IsDigit(thisCh), thatNumeric = char.IsDigit(thatCh);
 
-                while ((thisMarker < s1.Length) && (thisChunk.Length == 0 || SortHelper.InChunk(thisCh, thisChunk[0])))
+                while ((thisMarker < s1.Length) && (char.IsDigit(thisCh) == thisNumeric))
```
Comment (cvium) [inline MediaBrowser.Controller/Sorting/AlphanumComparator.cs]:
```suggestion
                while (thisMarker < s1.Length && char.IsDigit(thisCh) == thisNumeric)
```

File: MediaBrowser.Controller/Sorting/AlphanumComparator.cs [hunk 3]
```diff
@@ -43,7 +44,7 @@ public static int CompareValues(string s1, string s2)
                     }
                 }
 
-                while ((thatMarker < s2.Length) && (thatChunk.Length == 0 || SortHelper.InChunk(thatCh, thatChunk[0])))
+                while ((thatMarker < s2.Length) && (char.IsDigit(thatCh) == thatNumeric))
```
Comment (cvium) [inline MediaBrowser.Controller/Sorting/AlphanumComparator.cs]:
```suggestion
                while (thatMarker < s2.Length && char.IsDigit(thatCh) == thatNumeric)
```

File: MediaBrowser.Controller/Sorting/SortExtensions.cs [hunk 4]
```diff
@@ -7,137 +7,25 @@ namespace MediaBrowser.Controller.Sorting
 {
     public static class SortExtensions
     {
+        private static IComparer<string> _comparer = new AlphanumComparator();
```
Comment (Bond-009) [inline MediaBrowser.Controller/Sorting/SortExtensions.cs]:
```suggestion
        private static readonly AlphanumComparator<string> _comparer = new AlphanumComparator();
```
This doesn't need to be an interface, does it?

**Comment by JustAMan** [comment]
> I hope this got tested ;)

I took a leaf outta your book, also what can go wrong? 😆

File: MediaBrowser.Controller/Sorting/SortExtensions.cs (hunk 4 above)
Comment (JustAMan) [reply MediaBrowser.Controller/Sorting/SortExtensions.cs]: Yeah, it doesn't. Though `AlphanumComparator` isn't a template class :)

**Pull request merged**